Source/Provers/SMTLib/SMTLibProcess.cs

```diff
@@ -416,42 +405,18 @@ IEnumerable<SExpr> ParseSExprs(bool top)
 
     #region handling input from the prover
 
-    string ReadProver()
+    private readonly Queue<TaskCompletionSource<string>> outputReceivers = new();
+    
+    Task<string> ReadProver()
     {
-      string error = null;
-      while (true)
-      {
-        if (error != null)
-        {
-          HandleError(error);
-          errorCnt++;
-          error = null;
+      lock (this) {
```

Comment:
Do ou want external clients to be able take this lock?  If not I'd use a private object.

Reply:
Oh but I see below that the code was already like this, sorry

Reply:
Ah, good point. Never thought of that. Is it worth the extra code though? External clients taking the lock seems like only a theoretical issue.